public/js/reports/reports.service.js

```diff
@@ -79,5 +80,33 @@
 
             return columnDescription;
         };
+
+        function checkPrompts (prompts) {
+            const promptsStatus = {};
+            for (const p of prompts) {
+                let isValid;
+                if (p.elementType === 'date') {
+                    if (p.filterType === 'between') {
+                        isValid = !!(p.criterion.date1 && p.criterion.date2);
+                    } else if (p.filterType === 'equal-pattern') {
+                        isValid = (p.criterion.label && p.criterion.datePattern) !== '';
```

Comment:
Same problem than with dates: it will return true if one of them (or both) are undefined or null

Reply:
oups I forgot it